pkg/hubble/metrics/api/context.go

```diff
@@ -524,3 +525,26 @@ func (o *ContextOptions) Status() string {
 
 	return strings.Join(status, ",")
 }
+
+func (o *ContextOptions) DeleteMetricsAssociatedWithPod(name string, namespace string, vec *prometheus.MetricVec) {
+
+	sourceCtxCfg := append(o.Source, o.SourceEgress...)
+	sourceCtxCfg = append(sourceCtxCfg, o.SourceIngress...)
+	for _, contextID := range sourceCtxCfg {
+		if contextID == ContextPod {
```

Comment:
This should probably handle `ContextPodShort` and `ContextLabels`. For `ContextLabels` it should check if `source_pod`, or `destination_pod` values were configured, and if so, check their values.

Reply:
+1 for ContextLabels, but I believe it should only remove data series when both "source_pod+source_namespace" or "destination_pod+destination_namespace" is set.

I disagree with handling "ContextPodShort". There is a possibility that multiple pods from single deployment runs on a single k8s node. If we remove dataseries on pod deletion configured with ContextPodShort, we would remove it when one of the pods gets deleted and others are not affected. This will be definitely  counter-intuitive and not expected for users.

Reply:
Your right, PodShort is basically "WorkloadName", but not as reliable.